bench/sha256.bench.mo

```diff
@@ -0,0 +1,74 @@
+import Array "mo:base/Array";
+import Blob "mo:base/Blob";
+import Iter "mo:base/Iter";
+import Text "mo:base/Text";
+
+import Prim "mo:prim";
+
+import Bench "mo:bench";
+
+import Sha256 "../src/Sha256";
+
+module {
+  public func init() : Bench.Bench {
+    let bench = Bench.Bench();
+
+    bench.name("Sha256");
+    bench.description("Hash various message lengths from different types of input. Blocks are 64 bytes.");
+
+    let rows = [
+      "fromBlob",
+      "fromArray",
+      "fromIter",
+    ];
+    let cols = [
+      "0",
+      "1k blocks",
+      "1 MB",
+    ];
+
+    bench.rows(rows);
+    bench.cols(cols);
+
+    let rowSourceArrays : [[Nat8]] = [
+      [],
+      Array.tabulate<Nat8>(64_000, func(i) = 0x5f),
+      Array.tabulate<Nat8>(1_048_576, func(i) = 0x5f),
+    ];
+
+    let routines : [() -> ()] = Array.tabulate<() -> ()>(
+      rows.size() * cols.size(),
+      func(i) {
+        let row : Nat = i % rows.size();
+        let col : Nat = i / rows.size();
+
+        let source = rowSourceArrays[col];
+
+        switch (row) {
+          case (0) {
+            let blob = Blob.fromArray(source);
+            func() = ignore Sha256.fromBlob(#sha256, blob);
+          };
+          case (1) {
+            func() = ignore Sha256.fromArray(#sha256, source);
+          };
+          case (2) {
+            let iter = Iter.fromArray(source);
```

Comment:
This is insignificant amount of work. But just for consistency with the blob test above (for the reader) I would create this iter also outside the measured code.

Reply:
Got it. Thanks